### PR TITLE
feat: progress supports string argument

### PR DESCRIPTION
**Pull request by evilebottnawi**
**What kind of change does this PR introduce?**

feature

**Did you add tests for your changes?**

Yes

**If relevant, did you update the documentation?**

No

**Summary**

request from @sokra 

**Does this PR introduce a breaking change?**

No

**Other information**

No

### CODE REVIEW

File: packages/webpack-cli/lib/plugins/WebpackCLIPlugin.js [hunk 1]
```diff
@@ -18,11 +19,13 @@ class WebpackCLIPlugin {
...
                 if (compiler.options.plugins) {
-                    progressPluginExists = Boolean(compiler.options.plugins.find((e) => e instanceof ProgressPlugin));
+                    progressPluginExists = Boolean(compiler.options.plugins.find((plugin) => plugin instanceof ProgressPlugin));
                 }
 
                 if (!progressPluginExists) {
-                    new ProgressPlugin().apply(compiler);
+                    const isProfile = this.options.progress === 'profile';
+
+                    new ProgressPlugin({ profile: isProfile }).apply(compiler);
```
Comment (snitin315) [inline packages/webpack-cli/lib/plugins/WebpackCLIPlugin.js:28]: What if an invalid value is provided `webpack --progress=invalid`. Should we throw error ?

Comment (evilebottnawi) [reply packages/webpack-cli/lib/plugins/WebpackCLIPlugin.js:33]: Fixed, please review again

File: packages/webpack-cli/lib/plugins/WebpackCLIPlugin.js [hunk 2]
```diff
@@ -18,11 +19,18 @@ class WebpackCLIPlugin {
...
 
                 if (compiler.options.plugins) {
-                    progressPluginExists = Boolean(compiler.options.plugins.find((e) => e instanceof ProgressPlugin));
+                    progressPluginExists = Boolean(compiler.options.plugins.find((plugin) => plugin instanceof ProgressPlugin));
                 }
 
                 if (!progressPluginExists) {
-                    new ProgressPlugin().apply(compiler);
+                    if (typeof this.options.progress === 'string' && this.options.progress !== 'profile') {
+                        logger.error(`Invalid ${this.options.progress} value for the profile option. Allowed value is profile.`);
```
Comment (snitin315) [inline packages/webpack-cli/lib/plugins/WebpackCLIPlugin.js:27]:
```suggestion
                        logger.error(`Invalid ${this.options.progress} value for the progress option. Allowed value is profile.`);
```

**Review by snitin315 — APPROVED** [review]

File: test/progress/progress-flag.test.js [hunk 3]
```diff
@@ -4,17 +4,38 @@ const { run } = require('../utils/test-utils');
...
+        expect(stderr).toMatch(/\[webpack\.Progress] \d+ ms setup/);
         expect(stderr).toContain('[webpack.Progress] 100%');
         expect(stdout).toContain('main.js');
     });
 
+    it('should not support invalid value', () => {
+        const { stderr, stdout, exitCode } = run(__dirname, ['--progress=unknown']);
+
+        expect(exitCode).toBe(2);
+        expect(stderr).toContain('[webpack-cli] Invalid unknown value for the profile option. Allowed value is profile.');
```
Comment (snitin315) [inline test/progress/progress-flag.test.js:28]: CI seems unhappy 😄 

```suggestion
        expect(stderr).toContain('[webpack-cli] Invalid unknown value for the progress option. Allowed value is profile.');
```

**Review by snitin315 — CHANGES_REQUESTED** [review]

**Review by snitin315 — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Pull request merged**